web-client/public/js/graph.js

```diff
@@ -1,4 +1,5 @@
 import Grid from "d3-v4-grid";
+import { grnState } from "./grnstate";
```

Comment:
I see why it was necessary for you to do this, but let's make sure this is temporary. You treat `grnState` as read-only here, so that's good, in-keeping with the MVC flow. Eventually, that code should somehow make it to `updateApp` instead. But I can see from where it is used that this case is not as simple as the others, so this intermediate stage is OK.